### DISCUSSION TITLE
Getting started with Avalonia, a real frustration

### DISCUSSION THREAD

**Discussion by Skyhikeeper** [AvaloniaUI/Avalonia#10005] [General]
Hi I am totally new to UI dev , having come from a back office development team.
We are trying to see if avalonia will be our way forward but we having major issues getting started
We create the app on visual studio 2022 and run it it works displays the text, as soon as we add any control to the screen it fails
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="LPRS.BigScreen.MainWindow"
        Title="LPRS.BigScreen">
	<StackPanel Orientation="Vertical" HorizontalAlignment="">
		<TextBlock Text="testdfdsfsdfsd"/>
		<TextBox Text="some name"/>
		<button Content="SAY HI"/>
	</StackPanel>
</Window>
```
Now this shoudl not be difficult at all but we dont have an idea what the issue can be

**Comment by emmauss** [comment]
It seems there's an issue with markup in your message. The code snippet doesn't show up.
For reference sake, is this the snippet?
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="LPRS.BigScreen.MainWindow"
        Title="LPRS.BigScreen">
	<StackPanel Orientation="Vertical" HorizontalAlignment="">
		<TextBlock Text="testdfdsfsdfsd"/>
		<TextBox Text="some name"/>
		<button Content="SAY HI"/>
	</StackPanel>
</Window>
```
You haven't set the value of `HorizontalAlignment`, and instead left it as an empty string. Remove that attribute and it will render.
If you have little or no knowledge about xaml frameworks, you can try our tutorial here, https://docs.avaloniaui.net/tutorials/todo-list-app .

**Reply by timunie** [reply]
Let me add this: https://github.com/AvaloniaUI/Avalonia.Samples

**Comment by hez2010** [comment]
The first character of "button" in your code should be capitalized. Also you didn't set the value of `HorizontalAlignment`.

**Comment by christosk92** [comment]
Use rider for better IntelliSense support & better experience.
